scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryScioContext.scala

```diff
@@ -54,10 +54,39 @@ final class BigQueryScioContext(@transient val self: ScioContext) extends Serial
     self.read(BigQueryTable(tableSpec))
 
   /**
-   * Get a typed SCollection for a BigQuery SELECT query or table.
+   * Get an SCollection for a BigQuery table using the storage API.
+   *
+   * @param selectedFields names of the fields in the table that should be read. If empty, all
+   *                       fields will be read. If the specified field is a nested field, all the
+   *                       sub-fields in the field will be selected.
+   * @param rowRestriction SQL text filtering statement, similar ti a WHERE clause in a query.
+   *                       Currently, we support combinations of predicates that are a comparison
+   *                       between a column and a constant value in SQL statement. Aggregates are
+   *                       not supported. For example:
+   *
+   * {{{
+   * "a > DATE '2014-09-27' AND (b > 5 AND c LIKE 'date')"
+   * }}}
+   */
+  def bigQueryStorage(tableSpec: String,
+                      selectedFields: List[String] = Nil,
+                      rowRestriction: String = null): SCollection[TableRow] =
+    self.read(BigQueryStorage(tableSpec))(BigQueryStorage.ReadParam(selectedFields, rowRestriction))
+
+  /**
+   * Get an SCollection for a BigQuery table using the storage API.
+   */
+  def bigQueryStorage(table: TableReference,
+                      selectedFields: List[String],
```

Comment:
Why do we require this fields with `TableReference`?

Reply:
Because otherwise it'll conflict with overloaded `def bigQueryStorage(tableSpec, ...)` with default `null`s above.

Reply:
oh yes! good old overloading error due to default args! I think we should take a stance on this :smile: From an api pov this feels weird coz it's imposing requirements where they don't exist. One thing we can do is to move this default to a new type. wdyt?

```scala
def bigQueryStorage(table: TableReference): SCollection[TableRow]

def bigQueryStorage(table: TableReference, options: ReadOptions): SCollection[TableRow]
```

Reply:
This is not super user facing. Do it in a separate PR?